application/ui/components/params.jsx

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+var _               = require('underscore');
+var React           = require('react');
+var ParamRow        = require('./param-row');
+
+module.exports = React.createClass({
+    displayName : 'Params',
+
+    propTypes : {
+        params   : React.PropTypes.array,
+        onChange : React.PropTypes.func
+    },
+
+    getInitialState : function()
+    {
+        var values   = {};
+        var includes = {};
+
+        _.each(this.props.params, function(param) {
+            includes[param.name] = true;
+
+            if (typeof param.defaultValue !== 'undefined') {
+                values[param.name] = param.defaultValue;
+                return;
+            }
+
+            switch (param.type) {
+                case 'string':
+                    values[param.name] = '';
+                    break;
+                case 'number':
+                    values[param.name] = 0;
+                    break;
+                case 'boolean':
+                    values[param.name] = false;
+                    break;
+                case 'enum':
+                    values[param.name] = param.defaultValue;
+                    break;
+                case 'array':
+                    values[param.name] = [];
+                    break;
+                case 'object':
+                case 'hash':
+                case 'custom-object':
+                    values[param.name] = {};
+                    break;
+                case 'stripe-token':
+                    values[param.name] = null;
+                    break;
+            }
+        });
+
+        return {
+            values      : values,
+            nullKeys    : {},
+            includeKeys : includes
```

Comment:
Would it make more sense to hold `nullKeys` and `includeKeys` as arrays instead of objects? As far as I can see, they are just objects where the `param.name` key is `true`, which essentially makes them just a list.